### ISSUE TITLE
System.IO.Compression.ZipArchive hangs on a corrupt/invalid zip stream from Azure Storage

### ISSUE THREAD

**Issue by AbhishekTripathi** [dotnet/runtime#36355]
I have found that the `ZipArchive` goes in an infinite loop when a zip stream is provided from Azure Storage. Below is a code snippet which is easy to reproduce by using any arbitrary file and renaming to zip or using a corrupt zip itself. Using System.IO.Compression v4.3.0. Note that when the files exist on local disk, the behavior is as expected and I get proper exception.

```
var badBlobPath = @"https://<storage>.blob.core.windows.net/<container>/somefile.zip";
var blobStream = new Microsoft.Azure.Storage.Blob.CloudBlockBlob(new Uri(badBlobPath));
var stream = blobStream.OpenRead();
//below line goes in infinite loop.
var zipArchive = new ZipArchive(stream, ZipArchiveMode.Read, false);
```

**Comment by carlossanlop** [comment]
@AbhishekTripathi thanks for reporting. Do you have the callstack where the loop happens?

**Comment by AbhishekTripathi** [comment]
I may have to debug the nuget source but the function never returns. Seems it gets stuck in infinite loop.

**Comment by graviteja94** [comment]
Any update on the fix/alternative workaround for this issue?

**Comment by danmoseley** [comment]
Can you share callstack @graviteja94 ?

**Comment by graviteja94** [comment]
Hi @danmoseley 
Could you see the attached screenshot?
<img width="688" alt="image" src="https://user-images.githubusercontent.com/93539235/225359567-93b7165e-3c9c-43fc-80ca-8ae6759d166a.png">
upon executing the using statement, for a corrupted zip it never goes to the next line and the call stack is also empty 
<img width="649" alt="image" src="https://user-images.githubusercontent.com/93539235/225360242-c7f4a0cb-18ec-4556-b063-130d88284fdf.png">

basically my code keeps running indefinitely without throwing any exception.

I'm not sure if I provided the information you requested, can you point me out on how to collect the call stack?

**Comment by danmoseley** [comment]
Hi @graviteja94 I meant, break in at the point of the hang and get a stack. but if the hang is just looping in your code, of course that's not useful.

I'll let the area owners for compression speak to the issue as it's not my area

**Comment by jzirkelbach** [comment]
I'm currently attempting to read a corrupt zip file.  When manually testing the archive file by 7-Zip, it's identified with an `Unexpected end of data` and has a `Data error` for only one of its many files.  When attempting to access this zip file via ZipArchive, I'm observing an infinite loop at the following location:
https://github.com/dotnet/runtime/blob/main/src/libraries/System.IO.Compression/src/System/IO/Compression/ZipHelper.cs#L119-L139
with callstack
```
>	System.IO.Compression.dll!System.IO.Compression.ZipHelper.SeekBackwardsToSignature(System.IO.Stream stream, uint signatureToFind) Line 89	C#
 	System.IO.Compression.dll!System.IO.Compression.ZipArchive.ReadEndOfCentralDirectory() Line 390	C#
 	System.IO.Compression.dll!System.IO.Compression.ZipArchive.Init(System.IO.Stream stream, System.IO.Compression.ZipArchiveMode mode, bool leaveOpen) Line 340	C#
 	System.IO.Compression.dll!System.IO.Compression.ZipArchive.ZipArchive(System.IO.Stream stream, System.IO.Compression.ZipArchiveMode mode) Line 100	C#
```
![image](https://github.com/dotnet/runtime/assets/26876517/d77261df-73e3-44e8-a2e9-2d2ed2e83c01)

I hope this helps.

**Comment by danmoseley** [comment]
@jzirkelbach thanks, that does help. Any interest in investigating ?

**Comment by rzikm** [comment]
There were some recent changes around the area with the hang/livelock and I don't seem to be able to reproduce this with a fuzzer (neither on current main or in the version of the code at the time of filing this issue).

It would be helpful to get hold of a file which reproduces this issue, but otherwise I don't think this issue is actionable for us
